extension/zpagesextension/factory.go

```diff
@@ -54,24 +53,5 @@ func createExtension(_ context.Context, params component.ExtensionCreateParams,
 		return nil, errors.New("\"endpoint\" is required when using the \"zpages\" extension")
 	}
 
-	// The runtime settings are global to the application, so while in principle it
-	// is possible to have more than one instance, running multiple does not bring
-	// any value to the service.
-	// In order to avoid this issue we will allow the creation of a single
-	// instance once per process while keeping the private function that allow
-	// the creation of multiple instances for unit tests. Summary: only a single
-	// instance can be created via the factory.
-	if !atomic.CompareAndSwapInt32(&instanceState, instanceNotCreated, instanceCreated) {
```

Comment:
I think this comment was explaining nicely the motivation, why do you need to do it differently?

Reply:
The comment applies to pprof since it is the one that has runtime settings for the whole process. The comments for pprof were moved to the start method.